crates/abq_cli/src/main.rs

```diff
@@ -620,17 +654,45 @@ struct ResolvedConfig {
     queue_location: QueueLocation,
 }
 
+#[derive(Debug, PartialEq, Eq)]
 enum QueueLocation {
     Remote(SocketAddr),
     Ephemeral { opt_tls_key: Option<Vec<u8>> },
+    Unsupported(String),
 }
 
 impl QueueLocation {
     fn is_remote(&self) -> bool {
-        match self {
-            QueueLocation::Remote(_) => true,
-            QueueLocation::Ephemeral { .. } => false,
+        matches!(self, QueueLocation::Remote(_))
+    }
+}
+
+fn determine_queue_location(
+    access_token_kind: Option<AccessTokenKind>,
+    run_id_provided: bool,
+    queue_addr: Option<SocketAddr>,
+    usage_error_from_api: Option<String>,
+    tls_key: Option<Vec<u8>>,
+) -> QueueLocation {
```

Comment:
It might be more readable for the callers and tests to pass a config struct here, like

```
struct QueueLocationConfig { ... }

fn determine_queue_location(config: QueueLocationConfig)
```